app/uk/gov/hmrc/economiccrimelevyregistration/controllers/RegistrationSubmittedController.scala

```diff
@@ -38,7 +38,14 @@ class RegistrationSubmittedController @Inject() (
       .get(SessionKeys.EclReference)
       .getOrElse(throw new IllegalStateException("ECL reference number not found in session"))
 
-    Ok(view(eclReference))
+    val firstContactEmailAddress: String = request.session
+      .get(SessionKeys.FirstContactEmailAddress)
+      .getOrElse(throw new IllegalStateException("First contact email address not found in session"))
+
+    val secondContactEmailAddress: Option[String] =
+      request.session.get(SessionKeys.SecondContactEmailAddress).flatMap(Option(_).filter(_.trim.nonEmpty))
```

Comment:
if you do my suggestion in `CheckYourAnswersController` you can reduce this down to:

```suggestion
      request.session.get(SessionKeys.SecondContactEmailAddress)
```